### PR TITLE
[Editorial] Bracket Sequences II

### PR DESCRIPTION
**Pull request by brebenelmihnea**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/advanced/cses-2187.mdx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+## Explanation
+
+First, it would be helpful to know how many open and closed brackets should be in the remaining sequence; let $c$ and $n$
+denote the number of closed brackets and the length of the remaining sequence. Then, the number of possible endings -
+not necessarily valid - is ${n \choose c}$.
+
+Second, not all ${n \choose c}$ endings are valid, so by counting the invalid endings we'll know that the rest must be valid.
+To count the invalid endings, flip the brackets in the suffix beginning from the firt position the sequence becomes invalid,
+resulting in $c+1$ closed brackets, thus the number of invalid endings is ${n \choose c+1}$. The answer is the difference between
```
Comment (SansPapyrus683) [inline solutions/advanced/cses-2187.mdx:16]: https://en.wikipedia.org/wiki/Catalan_number#Second_proof
honestly this link explains it better

Comment (brebenelmihnea) [reply solutions/advanced/cses-2187.mdx:16]: Ok should I just add it or remove what I wrote ?

Comment (SansPapyrus683) [reply solutions/advanced/cses-2187.mdx:16]: second

Comment (brebenelmihnea) [reply solutions/advanced/cses-2187.mdx:16]: :((

Comment (brebenelmihnea) [reply solutions/advanced/cses-2187.mdx]: Ok I removed it and added the wiki

**Review by SansPapyrus683 — APPROVED** [review]
ねえ 女の子になりたい

**Review by ryanchou-dev — APPROVED** [review]
send ac screenshot and we're chilling

good work!

**Comment by brebenelmihnea** [comment]
https://cses.fi/problemset/result/10113988/
image

**Pull request merged**